crates/core/tedge_api/src/messages.rs

```diff
@@ -481,6 +534,31 @@ impl CommandPayload for RestartCommandPayload {
     }
 }
 
+#[derive(Debug, Clone, Deserialize, Serialize, Eq, PartialEq)]
+#[serde(rename_all = "camelCase")]
+struct RestartContext {
+    command: GenericCommandState,
+    on_success: StateName,
+    on_error: StateName,
+}
+
+impl RestartContext {
+    pub fn resume(&self, error: Option<String>) -> GenericCommandState {
+        let command = self.command.clone();
+        match error {
+            None => command.move_to(self.on_success.clone()),
+            Some(reason) => {
+                let command = command.update_from_json(json!({ "restartError": reason }));
+                if self.on_error == "failed" {
+                    command.fail_with(reason)
+                } else {
+                    command.move_to(self.on_error.clone())
+                }
```

Comment:
```suggestion
                if self.on_error == "failed" {
                    command.fail_with(reason)
                } else {
                    let command = command.update_from_json(json!({ "restartError": reason }));
                    command.move_to(self.on_error.clone())
                }
```

Reply:
Changed. Indeed better to not duplicate the error message.